loadflow/loadflow-api/src/main/java/com/powsybl/loadflow/LoadFlowProvider.java

```diff
@@ -43,15 +47,22 @@
      * to be stateless so that it can be call simultaneously with different arguments (a different network for instance)
      * without any concurrency issue.
      *
-     * @param network the network
+     * @param network            the network
      * @param computationManager a computation manager to external program execution
-     * @param workingVariantId variant id of the network
-     * @param parameters load flow execution parameters
-     * @param reporter the reporter used for functional logs
+     * @param workingVariantId   variant id of the network
+     * @param parameters         load flow execution parameters
+     * @param reporter           the reporter used for functional logs
      * @return a {@link CompletableFuture} on {@link LoadFlowResult]
      */
     default CompletableFuture<LoadFlowResult> run(Network network, ComputationManager computationManager, String workingVariantId, LoadFlowParameters parameters, Reporter reporter) {
         return run(network, computationManager, workingVariantId, parameters);
     }
 
+    // Defines the JSON serializer/deserializer to use
+    Optional<JsonLoadFlowParameters.ExtensionSerializer<? extends Extension<LoadFlowParameters>>> getParametersExtensionSerializer();
```

Comment:
You should use ExtensionJsonSerializer directly. JsonLoadFlowParameters.ExtensionSerializer<? extends Extension<LoadFlowParameters>> should be removed as it is a marker interface now (bad practice) and was here before only as a base interface for service loader (now removed).

Reply:
It's not only a marker interface, it's also a specialization (T = LoadFlowParameters) --> I don't think we should remove it!

Reply:
I never seen in Java, "typedef" like design pattern considered as a good practice.
When you have a complex type like Map<String, List<Foo>> you never create a class to simplify the typing (and Map is probably final to avoid this)

Reply:
Ok!